ledger/internal/evalprefetcher.go

```diff
@@ -279,18 +279,21 @@ func (p *accountPrefetcher) prefetch(ctx context.Context) {
 			case protocol.AssetConfigTx:
 				loadAccountsAddResourceTask(nil, basics.CreatableIndex(stxn.Txn.ConfigAsset), basics.AssetCreatable, task, resourceTasks, queue)
 			case protocol.AssetTransferTx:
-				loadAccountsAddResourceTask(&stxn.Txn.Sender, basics.CreatableIndex(stxn.Txn.XferAsset), basics.AssetCreatable, task, resourceTasks, queue)
 				if !stxn.Txn.AssetSender.IsZero() {
+					loadAccountsAddResourceTask(nil, basics.CreatableIndex(stxn.Txn.XferAsset), basics.AssetCreatable, task, resourceTasks, queue)
```

Comment:
in the case of `AssetTransferTx` with non-zero `AssetSender`, you've removed the loading of the `Txn.Sender` asset resource loading ( i.e. clawback ). I'm pretty sure that in case of a clawback, we need to load the sender resources ( so we can claw them back ).